src/sentry/users/api/endpoints/user_emails.py

```diff
@@ -78,7 +79,7 @@ def get(self, request: Request, user: User) -> Response:
 
         emails = user.emails.all()
 
-        return self.respond(serialize(list(emails), user=user))
+        return self.respond(serialize(list(emails), user=user, serializer=UserEmailSerializer()))
```

Comment:
Extremely weird bug and this change is one workaround.

For some reason none of the users/moved `serializers` were being registered in the registry despite having the register decorator. This also happened for the `authenticator` serializer when **only** it was moved. 

Some workarounds I found, 
- Import at least one users serializer in `sentry.api.serializers` `__init__.py` for some reason, being imported here will also cause the register to fire
- Import at least one users serializer in an endpoint (I assume it doesn't need to be an endpoint but just any file that is initialized ? I thought the `sentry.users.api.serializers` init.py would work, but I guess not?)
- The `UserSerializer` will always be registered for some reason, so if the other users serializers are also imported in the same `__init__.py`, they will all be registered. This is what happened for `authenticator` serializer in https://github.com/getsentry/sentry/pull/76896

Reply:
Note 2: I haven't tried recreating this in CI tests, but it was p. reproducible in local tests. E.g run this branch without `UserEmailSerializer` being imported/revert all changes in `User_Email` endpoint. Or just migrate any serializer (not UserSerializer) into users directory and run tests

Reply:
> For some reason none of the users/moved serializers were being registered in the registry despite having the register decorator. This also happened for the authenticator serializer when only it was moved.

This is likely because the new locations aren't being imported at all, so the registration calls never happen. We're likely relying on application startup importing `sentry.api.serializers` which loads all of the other serializer modules.

> On a separate note: it may be a good idea to explicitly pass in the specific serializer more, it helps prevent the above bug and is clearer

I agree. Being explicit and more obvious about which serializer an endpoint uses is clearer and helps us minimize the number of modules that are imported during application startup as well.

Reply:
Gotchas that makes sense. I think something that confused me was that `UserSerializer` would always get imported despite the move and I was struggling/didn't know how to trace backwards to figure out where it was getting imported from. 

I added some print statements yesterday and saw that the `__init__.py` for `sentry.users.api.serializers` was getting "run" (?) during the django setup but not registered (?)

Reply:
Okay, followup to this for closure, I think `UserSerializer` was always imported because `DetailedSelfUserSerializer` was being explicitly passed already in the `user_details` endpt